libres/tests/analysis/test_enkf_linalg.cpp

```diff
@@ -0,0 +1,68 @@
+#include "catch2/catch.hpp"
+
+#include "ert/analysis/enkf_linalg.hpp"
+
+/*
+Results generated by this equivalent implementation in numpy:
+
+import numpy as np
+nrow = 2
+ncol = 3
+
+S = np.ones((nrow, ncol))
+S[0, 1] = 1.5
+S[1, 0] = 2.5
+
+W = np.ones((nrow, ncol))
+W[0, 1] = 1.3
+
+eig = np.array([2.0, 2.0, 3.0])
+
+X2 = (W.T @ S)
+X2 = (np.sqrt(eig) * X2.T).T
+X2
+*/
+double *test_enkf_linalg_genX2() {
```

Comment:
This allocates but doesn't free. Maybe you can move this into the test and use BDD?

```c++
TEST_CASE(...) {
    GIVEN("a matrix") {
        // Matrix alloc
        THEN("Use enkf_linalg_genX2") {
           enkf_linalg_genX2(...);
           REQUIRE(...);
        }
        // Matrix free
    }
}
```

Reply:
Is it enough to release `S` and `W`?

```C
    matrix_free(S);
    matrix_free(W);
```

Reply:
You need to free `X2` also. Maybe you can change the later code to:
```cpp
    auto matrix = test_enkf_linalg_genX2();
    auto result = matrix->data;

    // ...

    matrix_free(matrix);
```